### PR TITLE
Feature trip java to kotlin

### LINKED ISSUE #5: Pakiet trips - zmiana kodu z Javy na Kotlina

### PR DESCRIPTION
**Pull request by Alesta00103**

### CODE REVIEW

File: app/src/main/java/com/aleksandra/go4mytrip/AddNewTrip.kt [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+            sendTripInfoIntent.putExtra("timeStart", timeS)
+            sendTripInfoIntent.putExtra("timeEnd", timeE)
+            setResult(RESULT_OK, sendTripInfoIntent)
+            finish()
+        }
+    }
+
+    @SuppressLint("SetTextI18n")
+    private fun setViewOrUpdateTrip() {
+        tripName!!.setText(alreadyAvailableTrip!!.title)
```
Comment (marcin-prpm) [inline app/src/main/java/com/aleksandra/go4mytrip/AddNewTrip.kt]: you should avoid using the '!!' operator

File: app/src/main/java/com/aleksandra/go4mytrip/AddNewTrip.kt [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+        placeName!!.text = alreadyAvailableTrip!!.namePlace
+        dateOfTripStart!!.text = alreadyAvailableTrip!!.tripDate + " " + alreadyAvailableTrip!!.timeStart
+        dateS = alreadyAvailableTrip!!.tripDate
+        endDate!!.text = alreadyAvailableTrip!!.tripDateEnd + " " + alreadyAvailableTrip!!.timeEnd
```
Comment (marcin-prpm) [inline app/src/main/java/com/aleksandra/go4mytrip/AddNewTrip.kt]: here you can use kotlin string interpolation: https://kotlinlang.org/docs/java-to-kotlin-idioms-strings.html#concatenate-strings

File: app/src/main/java/com/aleksandra/go4mytrip/AddNewTrip.kt [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+        dateE = alreadyAvailableTrip!!.tripDateEnd
+        coordinates = alreadyAvailableTrip!!.coordinate
+        // dateOfTripEnd = alreadyAvailableTrip.getTripDateEnd();
```
Comment (marcin-prpm) [inline app/src/main/java/com/aleksandra/go4mytrip/AddNewTrip.kt]: please try not to commit code that is commented out

File: app/src/main/java/com/aleksandra/go4mytrip/AddNewTrip.kt [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+                placeName!!.text = placeNameText
+                coordinates = data.getStringExtra("coordinate")
+                Toast.makeText(this, "Added location", Toast.LENGTH_SHORT).show()
+            }
+        }
+    }
+
+    private fun selectImage() {
+        val intent = Intent(Intent.ACTION_PICK, MediaStore.Images.Media.EXTERNAL_CONTENT_URI)
+        if (intent.resolveActivity(packageManager) != null) {
```
Comment (marcin-prpm) [inline app/src/main/java/com/aleksandra/go4mytrip/AddNewTrip.kt]: it'd be a bit cleaner if you used 'let'

File: app/src/main/java/com/aleksandra/go4mytrip/AddNewTrip.kt [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+                selectImage()
+            } else {
+                Toast.makeText(this, "Permission Denied!", Toast.LENGTH_SHORT).show()
+            }
+        }
+    }
+
+    override fun onDateSet(view: android.widget.DatePicker, year: Int, month: Int, dayOfMonth: Int) {
+        if (isDataStart) {
+            calendarS = Calendar.getInstance()
```
Comment (marcin-prpm) [inline app/src/main/java/com/aleksandra/go4mytrip/AddNewTrip.kt]: you should extract two separate methods from this if - else block

File: app/src/main/java/com/aleksandra/go4mytrip/TripAdapter.kt [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+            val id = mData[position]!!.tripId
+            val title = mData[position]!!.title
+            val image = mData[position]!!.imageTrip
+            val dateS = mData[position]!!.tripDate
+            val dateE = mData[position]!!.tripDateEnd
+            val timeStart = mData[position]!!.timeStart
+            val timeEnd = mData[position]!!.timeEnd
+            val intent = Intent(context, DetailTrip::class.java)
+            intent.putExtra("idTripToDetail", id)
+            intent.putExtra("title", title)
```
Comment (marcin-prpm) [inline app/src/main/java/com/aleksandra/go4mytrip/TripAdapter.kt]: replace these fields with a dedicated model that implements the Parcelable interface

File: app/src/main/res/layout/activity_trips.xml [hunk 7]
```diff
@@ -57,9 +58,9 @@
                 android:textColor="@color/gray"
                 android:textColorHint="@color/hintTextSearch"
                 android:textSize="16sp"
```
Comment (marcin-prpm) [inline app/src/main/res/layout/activity_trips.xml]: all dimensions should be moved to dimens.xml

File: app/src/main/java/com/aleksandra/go4mytrip/Trips.kt [hunk 8]
```diff
@@ -0,0 +1,371 @@
...
+        referenceNotes = FirebaseDatabase.getInstance().getReference("notes")
+        referencePlaces = FirebaseDatabase.getInstance().getReference("placesToVisit")
+
+        inputSearch.addTextChangedListener(object : TextWatcher {
+            override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {}
+            override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {
+                referenceTrips.child(uid).addValueEventListener(object : ValueEventListener {
+                    override fun onDataChange(dataSnapshot: DataSnapshot) {
+                        tripList.clear()
+                        for (tripSnapshot in dataSnapshot.children) {
```
Comment (marcin-prpm) [inline app/src/main/java/com/aleksandra/go4mytrip/Trips.kt]: forEach

File: app/src/main/res/layout/activity_details_trip.xml [hunk 9]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 
-<RelativeLayout
+<androidx.constraintlayout.widget.ConstraintLayout
```
Comment (Alesta00103) [inline app/src/main/res/layout/activity_details_trip.xml]: Tutaj po zmianie z relativeLayout na ConstraintLayout chowa mi się część layoutu pod spód. Nie widać go, ale przyciski działają (packing list i shopping list), więc nie do końca wiem, co się dzieje.

**Pull request merged**